services/redis/queue.js

```diff
@@ -57,15 +59,27 @@ const registerV2ServiceQueue = async (token, userId, follows, topics, locations,
   }
 };
 
-const followTopicServiceQueue = async (user_id, topic_id) => {
+const followTopicServiceQueue = async (user_id, topic_id, community_message_format_id, delay) => {
   const data = {
     user_id,
-    topic_id
+    topic_id,
+    community_message_format_id
   };
+  let currentTime = momentTz().tz('America/Los_Angeles');
+  const randomTime = sample([6, 7, 8]);
+  const additionalDays = delay;
+
+  let requiredTime = momentTz()
+    .tz('America/Los_Angeles')
+    .set({hour: randomTime})
+    .add(additionalDays, 'days');
+
+  const diffTime = requiredTime.diff(currentTime, 'milliseconds');
 
   const options = {
     jobId: uuidv4(),
-    removeOnComplete: true
+    removeOnComplete: true,
+    delay: diffTime
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for calculating the delay time seems to be correct. However, there is no error handling in case the `followTopicQueue.add(data, options)` function fails. It would be better to add a catch block to handle any potential errors.

```diff
   try {
     await followTopicQueue.add(data, options);
+  } catch (error) {
+    console.error(`Error adding job to queue: ${error}`);
+    throw error;
   }
```

Also, it's important to note that using `console.error` is just a placeholder for actual error handling logic. In a production environment, you might want to use a logging service or some other form of error reporting mechanism.


<!-- This is an auto-generated comment by OSS CodeRabbit -->